src/Diagnostics.RuntimeHost/Middleware/DiagnosticsRequestMiddleware.cs

```diff
@@ -29,6 +33,12 @@ public async Task Invoke(HttpContext httpContext)
             {
                 await _next(httpContext);
             }
+            catch (TimeoutException ex)
+            {
+                httpContext.Response.Clear();
```

Comment:
You'll get an exception here if the Response has Stated. Something like "it cannot be modified at this time". So you should check if the response has not started and then update it.